connector/src/yang/connector/gnmi.py

```diff
@@ -150,6 +139,35 @@ def stopped(self):
         return self._stop_event.is_set()
 
 
+class CiscoAuthPlugin(grpc.AuthMetadataPlugin):
+    """A plugin which adds username/password metadata to each call."""
+
+    def __init__(self, username, password):
+        super(CiscoAuthPlugin, self).__init__()
+        self.username = username
+        self.password = password
+
+    def __call__(self, context, callback):
+        callback(
+            [("username", self.username), ("password", self.password)],
+            None
+        )
+
+
+class GnmiLogHandler(logging.Handler):
+
+    @property
+    def gnmi_session(self):
+        return self._gnmi_session
+
+    @gnmi_session.setter
+    def gnmi_session(self, session):
+        self._gnmi_session = session
+
+    def emit(self, record):
+        self.gnmi_session.results.append(record.msg)
+
+
 class Gnmi(BaseConnection):
     """Session handling for gNMI connections.
```

Comment:
You need to fix the class docstring.  Just give a working example of retrieving capabilities.  Try this out yourself to make sure it works.
```
    >>> from pyats.topology import loader
    >>> from yang.connector.gnmi import Gnmi
    >>> testbed=loader.load('testbed_native_test.yaml')
    >>> device=testbed.devices['ddmi-9500-2']
    >>> device.connect(alias='gnmi', via='yang2')
    >>> #####################
    >>> # Capabilities      #
    >>> #####################
    >>> resp=device.gnmi.capabilities()
    >>> print(resp)
```
Would be good to go further, you could give a simple "get" example.
```
from . import proto
request = proto.gnmi_pb2.GetRequest()
request.type = proto.gnmi_pb2.GetRequest.DataType.Value('ALL')
request.encoding = proto.gnmi_pb2.Encoding.Value('JSON)
request.path = proto.gnmi.Path()
....
resp = device.gnmi.get(request)
print(resp)
```